ast/src/parsed/types.rs

```diff
@@ -405,6 +405,12 @@ impl From<Type> for TypeScheme {
     }
 }
 
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Serialize, Deserialize, JsonSchema)]
+pub struct TraitScheme {
```

Comment:
I'm wondering (really wondering, not suggesting to do this) if it would be simpler to use a regular TypeScheme whose type is a tuple instead.

Reply:
I didn't really think about this solution, but it looks really more elegant than the current option. I'll take a look at it